File: BasicMTGScraper/src/main/java/com/wimpy/dao/entity/MtgCard.java
package com.wimpy.dao.entity;

import javax.persistence.*;

@Entity
@Table
public class MtgCard {


    @Id
    @GeneratedValue
    private long id;

    @Column
    private String name;

    public long getId() {
        return id;
    }

    public MtgCard setId(long id) {
        this.id = id;
        return this;
    }

    public String getName() {
        return name;
    }

    public MtgCard setName(String name) {
        this.name = name;
        return this;
    }
}

File: BasicMTGScraper/src/main/java/com/wimpy/dao/entity/MtgCard.kt
package com.wimpy.dao.entity;

import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import javax.persistence.*;
import java.util.Date;

@Entity
@Table
public class MtgCard {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    @Column
    private String name;

    @CreationTimestamp
    @Temporal(TemporalType.TIMESTAMP)
    private Date created;

    @UpdateTimestamp
    @Temporal(TemporalType.TIMESTAMP)
    private Date updated;

    public long getId() {
        return id;
    }

    public MtgCard setId(long id) {
        this.id = id;
        return this;
    }

    public String getName() {
        return name;
    }

    public MtgCard setName(String name) {
        this.name = name;
        return this;
    }

    public Date getCreated() {
        return created;
    }

    public void setCreated(Date created) {
        this.created = created;
    }

    public Date getUpdated() {
        return updated;
    }

    public void setUpdated(Date updated) {
        this.updated = updated;
    }
}
